### PR TITLE
Fix order.change_status creating empty history entries

### LINKED ISSUE #1229: Fix order history comments
Call of order.change_status generates history entries without comment.

### PR DESCRIPTION
**Pull request by koradon**
Change order.change_status to accept comment param.
Add comments to calls of order.change_status.

Fixes #1229

### CODE REVIEW

File: saleor/checkout/views/summary.py [hunk 1]
```diff
@@ -20,7 +21,8 @@ def create_order(checkout):
         return None, redirect('checkout:summary')
     checkout.clear_storage()
     checkout.cart.clear()
-    order.create_history_entry()
+    order.create_history_entry(status=OrderStatus.NEW, comment=pgettext_lazy(
```
Comment (elwoodxblues) [inline saleor/checkout/views/summary.py]: Shouldn't it also set the user who placed the order?

File: saleor/order/views.py [hunk 2]
```diff
@@ -100,6 +100,10 @@ def start_payment(request, order, variant):
             defaults=defaults)
         try:
             form = payment.get_form(data=request.POST or None)
+            order.create_history_entry(
```
Comment (elwoodxblues) [inline saleor/order/views.py]: I know we used to create an empty entry when starting a new payment, but I'm really guessing if this entry does help with anything.

Comment (koradon) [reply saleor/order/views.py]: I have the same thoughts with this one. I don't really know. Maybe in some cases but feels like hidden debug message?

File: saleor/checkout/views/summary.py (hunk 1 above)
Comment (koradon) [reply saleor/checkout/views/summary.py]: Yes, added

**Pull request merged**